src/components/contentAvail/contentAvailSrv.js

```diff
@@ -1,160 +1,230 @@
+                    };
+                    return StorageService.get(purchaseDataPath,defValues);
                 }
-                return _baseIsEntityAvail().then(function(res){
-                    if(res === true){
-                        return true;
-                    }
 
-                    var purchaseData = res[0];
-                    var freeContent = res[1];
+                function getFreeContentData(){
+                    var StorageService = InfraConfigSrv.getStorageService();
+                    var freeContentPath = 'freeContent';
+                    var defValues = {
+                        daily: 0,
+                        exam: {},
+                        tutorial: {},
+                        specials: {}
+                    };
+                    return StorageService.get(freeContentPath,defValues);
+                }
 
-                    if(freeContent.daily >= dailyOrder){
-                        return true;
-                    }
+                function getUserSpecialsData(){
+                    var specialsProms = {};
+                    angular.forEach(_specials, function(specialFn, key) {
+                        specialsProms[key] = $injector.invoke(specialFn);
+                    });
+                    return $q.all(specialsProms);
+                }
 
-                    if(angular.isString(purchaseData.daily)){
-                        return purchaseData.daily === PURCHASED_ALL;
-                    }else{
-                        var maxAvailDailyOrder = (purchaseData.daily || 0) + (freeContent.daily || 0);
-                        return dailyOrder <= maxAvailDailyOrder;
-                    }
-                });
-            };
+                function idToKeyInStorage(id){
+                    return 'id_' + id;
+                }
 
-            this.isExamAvail = function(examId){
-                return _baseIsEntityAvail().then(function(res){
-                    if(res === true){
-                        return true;
-                    }
+                function _hasSubscription(subscriptionObj){
+                    return subscriptionObj && subscriptionObj.expiryDate && subscriptionObj.expiryDate > Date.now();
+                }
 
-                    var purchaseData = res[0];
-                    var freeContent = res[1];
+                function _baseIsEntityAvail(){
+                    return $q.all([getUserPurchaseData(),getFreeContentData()]).then(function(res){
+                        var purchaseData = res[0];
+                        var hasSubscription = _hasSubscription(purchaseData.subscription);
+                        if(hasSubscription){
+                            return true;
+                        }else{
+                            if(!_specials) {
+                                return res;
+                            }
+                            return getUserSpecialsData().then(function(specialsRes) {
```

Comment:
added it, now the earnedSpecials is in res[3], changed it in isExamAvail and etc.